src/components/dialog/dialog.js

```diff
@@ -653,17 +657,24 @@ function MdDialogProvider($$interimElementProvider) {
       disableParentScroll: true,
       autoWrap: true,
       fullscreen: false,
+      zIndex: 80,
       transformTemplate: function(template, options) {
+
+        template = prepareTemplate(template);
+
         // Make the dialog container focusable, because otherwise the focus will be always redirected to
         // an element outside of the container, and the focus trap won't work probably..
         // Also the tabindex is needed for the `escapeToClose` functionality, because
         // the keyDown event can't be triggered when the focus is outside of the container.
-        return '<div class="md-dialog-container" tabindex="-1">' + validatedTemplate(template) + '</div>';
+        return '' +
+          '<div class="md-dialog-container" tabindex="-1" style="z-index: ' + options.zIndex +'">' +
+            template +
+          '</div>';
 
         /**
-         * The specified template should contain a <md-dialog> wrapper element....
+         * The specified template should contain a <md-dialog> wrapper element.
          */
-        function validatedTemplate(template) {
+        function prepareTemplate(template) {
           if (options.autoWrap && !/<\/md-dialog>/g.test(template)) {
```

Comment:
It's not something that was added in the PR, but it could be a good chance to change the regex with a faster `indexOf`.